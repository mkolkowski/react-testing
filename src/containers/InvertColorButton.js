import { connect } from 'react-redux'

import { colorInverse } from '../actions'
import CustomButton from '../components/CustomButton'

const mapStateToProps = (state, ownProps) => {
    return Object.assign({},state.colorReductor);
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        onClick: () => dispatch(colorInverse())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CustomButton)
