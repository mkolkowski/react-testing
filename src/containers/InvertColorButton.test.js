import React from 'react';
import InvertColorButton from './InvertColorButton';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

test('renders correctly', () => {
    //Arrange
    const mockStore = configureStore();
    let store = mockStore({ value: false });

    //Act
    const component = renderer.create(
        <InvertColorButton store={store} />
    );
    let tree = component.toJSON();

    //Assert
    expect(tree).toMatchSnapshot();
});

describe('behave correctly', () => {
    var getComponent = (initValue) =>{
        const mockStore = configureStore();
        let store = mockStore({ value: initValue });
        const component = renderer.create(
            <InvertColorButton store={store} />
        );
        return component;
    }

    test('value inverted to true', () => {
        //Arrange
        const component = getComponent(false);

        //Act
        component.root.findByType("button").props.onClick();

        //Assert
        expect(component.root.findByType("input").props.value, true);
    });

    test('value inverted to false', () => {
       //Arrange
       const component = getComponent(true);

       //Act
       component.root.findByType("button").props.onClick();

       //Assert
       expect(component.root.findByType("input").props.value, false);
    });
}
)
