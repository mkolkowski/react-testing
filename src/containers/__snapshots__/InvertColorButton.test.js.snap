// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    type="hidden"
    value={undefined}
  />
  <button
    onClick={[Function]}
  >
    Stan przycisku: 
    NON
  </button>
</div>
`;
